src/main/java/thePackmaster/cards/intriguepack/Favor.java

```diff
@@ -24,18 +20,14 @@ public Favor() {
     }
 
     public void use(AbstractPlayer p, AbstractMonster m) {
-        // List of rares.
-        List<AbstractCard> eligibleCards = Wiz.getCardsMatchingPredicate(c -> c.rarity == CardRarity.RARE && !c.hasTag(AbstractCard.CardTags.HEALING));
-
-        // Add one of them to hand.
-        if (eligibleCards.size() > 0) {
-            AbstractCard cardy = eligibleCards.get(AbstractDungeon.cardRandomRng.random(0, eligibleCards.size() - 1)).makeCopy();
-            cardy.setCostForTurn(0);
-            CardModifierManager.addModifier(cardy, new ExhaustMod());
-            CardModifierManager.addModifier(cardy, new EchoedEtherealMod());
-            CardModifierManager.addModifier(cardy, new GlowEchoMod());
-            Wiz.atb(new MakeTempCardInHandAction(cardy, true));
-        }
+        // Wow, so much easier once I found this lmao.
+        CardGroup cards = JediUtil.filterCardsForDiscovery(c -> !c.hasTag(CardTags.HEALING) && c.rarity == CardRarity.RARE);
```

Comment:
This has a difference that I don't think you want -- it gets rares of all colors and packs, which isn't how the previous code worked.

Here's what you can do to only get the right cards:
```suggestion
        List<AbstractCard> eligibleCards = Wiz.getCardsMatchingPredicate(c -> c.rarity == CardRarity.RARE && !c.hasTag(AbstractCard.CardTags.HEALING));
        CardGroup cards = new CardGroup(CardGroup.CardGroupType.UNSPECIFIED);
        cards.group.addAll(eligibleCards);
```

A few more lines, but not that bad.